### DISCUSSION TITLE
Use next/link as to change the url along with baseurl

### DISCUSSION THREAD

**Discussion by foramvora** [vercel/next.js#18851] [Help]
I have the following use case--

basepath -- /_foo/

Actual Url --
abc.com/_foo/post

Required Url -- 
abc.com/post

Can i use the `as` prop to change the full url along with basepath? As per the docs, its an optional decorator only for the relative path

**Comment by MiguelMachado-dev** [comment]
If you have the path `/test/final` you can try `<Link href="/test/final" as="/final">` and for server-side rendering just add this below code to your next.config.js;

```js
experimental: {
      async rewrites() {
        return [
          { source: '/final', destination: '/test/final' }
          ];
       }
    }
```

**Reply by foramvora** [reply]
@MiguelMachado-dev
i do not have /test in the actual source... its the basepath added in next.config.js as i need all the resources with /test/ in the url for a particular reason.

On using <Link href="/xzy/final" as="/final">, next appends the domain+basepath and url becomes : abc.com/_test_/final where test is **not** intended in the url...
pages directory structure is -- pages/xyz/final.js

**Comment by fofiu** [comment]
@foramvora did you ever solve this? I'm running into the same scenario as what you described in your original post.
